### DISCUSSION TITLE
Bad cache load destroying instance

### DISCUSSION THREAD

**Discussion by AjaniBilby** [requarks/wiki#6227] [Error / Bug Report]
I've reinstalled wikijs multiple times and confirmed every time I include my db.sqlite file I get the error
```
EACCES: permission denied, open '/data/cache/b29b5d2ce62e55412776ab98f05631e0aa96597b.bin'
```
That file does not exist, and because of that I can't load a single page from the wiki. Why does failing to load a cache file brick an entire Wikijs instance? It should just drop the cache if it fails to load it and rebuild.

**Comment by AjaniBilby** [comment]
*I'm currently using SQL lite to try and manually remove any references to a cache to get my wiki back and running for my work*

Update:
Can't seem to easily find where he reference to this cache is, so I've started manually scrapping the pages out so I can at least have something to help the other staff work.

**Reply by lal309** [reply]
I'm having the exact same problem. I was running 2.5.301 just fine and then suddenly the permission error showed up, saw that they was a new release so updated to that (2.5.303) but problem still persist. I thought, well its just a cache file so let me stop the container, delete the cache file from the local host directory and try again. No luck. My entire instance is bricked over a cache file... 

Did you solve your issue? If so, how? I'm using MariaDB with Git storage back end, Debian 12, WikiJS and Mariadb containerized.

**Reply by NGPixel** [reply]
Make sure `/data/cache` is writeable by wiki.js?

**Comment by AjaniBilby** [comment]
Here's a nice script to copy all of your pages out of the database in case of emergency:
```python
import sqlite3
import os

conn = sqlite3.connect('db.sqlite')
c = conn.cursor()
c.execute("SELECT path, render FROM Pages")

for row in c.fetchall():
  path = "./export/"+os.path.dirname(row[0])
  if path != "" and not os.path.exists(path):
    os.makedirs(path)

  with open("./export/"+row[0]+".html", 'w', encoding='utf-8') as f:
    f.write(row[1])

conn.close()
```
